modules/Menuing.py

```diff
@@ -0,0 +1,584 @@
+                struct.unpack('<I', ReadSymbol('sMonSummaryScreen'))[0], offset=0x40C4, size=2))[0]]
+        case 'Pokémon FireRed' | 'Pokémon LeafGreen':
+            return moves_list[struct.unpack('<H', ReadSymbol('gMoveToLearn'))[0]]
+        case 'Pokémon Ruby' | 'Pokémon Sapphire':
+            return moves_list[int.from_bytes(ReadSymbol('gMoveToLearn', size=1))]
+
+
+def GetMoveLearningCursorPos() -> int:
+    """
+    helper function that returns the position of the move learning cursor
+    """
+    match mGBA.game:
+        case 'Pokémon Emerald':
+            return int.from_bytes(ReadMemory(
+                struct.unpack('<I', ReadSymbol('sMonSummaryScreen'))[0], offset=0x40C6, size=1))
+        case 'Pokémon FireRed' | 'Pokémon LeafGreen':
+            return int.from_bytes(ReadSymbol('sMoveSelectionCursorPos'))
+        case 'Pokémon Ruby' | 'Pokémon Sapphire':
+            return int.from_bytes(ReadSymbol('gSharedMem', offset=0x18079, size=1))
+
+
+def CheckForPickup() -> NoReturn:
+    """
+    Function that handles pickup farming.
+    """
+    try:
+        pickup_threshold = config_battle['pickup_threshold']
+    except:
+        pickup_threshold = 1
+    pokemon_with_pickup = 0
+    pokemon_with_pickup_and_item = []
+    party = GetParty()
+    for i in range(len(party)):
+        if party[i]['ability'] == 'Pickup':
+            pokemon_with_pickup += 1
+            if party[i]['item']['name'] != 'None':
+                pokemon_with_pickup_and_item.append(i)
+    if pickup_threshold > pokemon_with_pickup > 0:
+        pickup_threshold = pokemon_with_pickup
+    if len(pokemon_with_pickup_and_item) >= pickup_threshold:
+        console.print('Pickup threshold is met! Gathering items.')
+        TakePickupItems(pokemon_with_pickup_and_item)
+
+
+def TakePickupItems(pokemon_indices: list):
+    """
+    Function that takes items from Pokémon that have the Pickup ability.
+
+    :param pokemon_indices: The list of indices representing the Pokémon to take items from.
+    """
+    NavigateStartMenu(1)
+    while not PartyMenuIsOpen():
+        PressButton(['A'])
+    for idx in pokemon_indices:
+        while GetPartyMenuCursorPos()['slot_id'] != idx:
+            if GetPartyMenuCursorPos()['slot_id'] > idx:
+                PressButton(['Up'])
+            else:
+                PressButton(["Down"])
+        if mGBA.game in ['Pokémon Emerald', 'Pokémon FireRed', 'Pokémon LeafGreen']:
+            while 'Choose a' in DecodeString(ReadSymbol('gStringVar4')):  # TODO English only
+                PressButton(['A'])
+            while 'Do what with' in DecodeString(ReadSymbol('gStringVar4')) and not 'an item?' in \
+                                    DecodeString(ReadSymbol('gStringVar4')):  # TODO English only
+                NavigateMenu(2)
+            while 'Do what with an' in DecodeString(ReadSymbol('gStringVar4')):  # TODO English only
+                NavigateMenu(1)
```

Comment:
I have noticed that any mons with a HM such as Fly will attempt to use the move instead of selecting 'Switch'.

Might be able to select 'Switch' from the bottom of the list or add + x known HMs to the menu index?